specutils/spectra/xrayspectrum1d.py

```diff
@@ -0,0 +1,573 @@
+            The upper edges of the energy bins
+
+        energ_unit : astropy.units.Unit
+            Description of the energy units used
+
+        detchans : int
+            The number of channels in the detector
+
+        """
+        self.filename = filename
+        self.offset = None
+        self.n_grp = None
+        self.f_chan = None
+        self.n_chan = None
+        self.matrix = None
+        self.energ_lo = None
+        self.energ_hi = None
+        self.energ_unit = None
+        self.detchans = None
+        self._load_rmf(filename)
+
+    def _load_rmf(self, filename, extension=None):
+        # open the FITS file and extract the MATRIX extension
+        # which contains the redistribution matrix and
+        # anxillary information
+        hdulist = fits.open(filename)
+
+        # get all the extension names
+        extnames = np.array([h.name for h in hdulist])
+
+        # figure out the right extension to use
+        if extension is not None:
+            h = hdulist[extension]
+
+        elif "MATRIX" in extnames:
+            h = hdulist["MATRIX"]
+
+        elif "SPECRESP MATRIX" in extnames:
+            h = hdulist["SPECRESP MATRIX"]
+
+        else:
+            print("Cannot find common FITS file extension for response matrix values")
+            print("Please set the `extension` keyword")
+            return
+
+        data = h.data
+        hdr = h.header
+        hdulist.close()
+
+        # extract + store the attributes described in the docstring
+        n_grp = np.array(data.field("N_GRP"))
+        f_chan = np.array(data.field('F_CHAN'))
+        n_chan = np.array(data.field("N_CHAN"))
+        matrix = np.array(data.field("MATRIX"))
+
+        self.energ_lo = np.array(data.field("ENERG_LO"))
+        self.energ_hi = np.array(data.field("ENERG_HI"))
+        self.energ_unit = _unit_parser(data.columns["ENERG_LO"].unit)
+        self.detchans = hdr["DETCHANS"]
+        self.offset = self.__get_tlmin(h)
+
+        # flatten the variable-length arrays
+        self.n_grp, self.f_chan, self.n_chan, self.matrix = \
+                self._flatten_arrays(n_grp, f_chan, n_chan, matrix)
+
+        return
+
+    def __get_tlmin(self, h):
+        """
+        Get the tlmin keyword for `F_CHAN`.
+
+        Parameters
+        ----------
+        h : an astropy.io.fits.hdu.table.BinTableHDU object
+            The extension containing the `F_CHAN` column
+
+        Returns
+        -------
+        tlmin : int
+            The tlmin keyword
+        """
+        # get the header
+        hdr = h.header
+        # get the keys of all
+        keys = np.array(list(hdr.keys()))
+
+        # find the place where the tlmin keyword is defined
+        t = np.array(["TLMIN" in k for k in keys])
+
+        # get the index of the TLMIN keyword
+        tlmin_idx = np.hstack(np.where(t))[0]
+
+        # get the corresponding value
+        tlmin = np.int(list(hdr.items())[tlmin_idx][1])
+
+        return tlmin
```

Comment:
I'm confused. Can't you get the same output as this function simply with `return int(h.header['TLMIN'])`?